src/main/java/uk/ac/cam/cl/dtg/segue/api/managers/GroupManager.java

```diff
@@ -410,6 +420,50 @@ public UserGroupDTO addUserToManagerList(final UserGroupDTO group, final Registe
         return this.getGroupById(group.getId());
     }
 
+    /**
+     * Transfer group ownership from the group owner to another user.
+     *
+     * @param group - group to affect
+     * @param newOwner - user to promote to owner of the group
+     * @param oldOwner - user (must be previous group owner) to demote to additional manager status
+     * @return The group DTO
+     * @throws SegueDatabaseException if there is a db error
+     * @throws IllegalAccessException if oldOwner is not the current owner of the group
+     */
+    public UserGroupDTO promoteUserToOwner(final UserGroupDTO group, final RegisteredUserDTO newOwner, final RegisteredUserDTO oldOwner) throws SegueDatabaseException, IllegalAccessException {
+        Validate.notNull(group);
+        Validate.notNull(newOwner);
+        Validate.notNull(oldOwner);
+
+        // Old owner must actually be the old (current) owner of the group
+        if (!oldOwner.getId().equals(group.getOwnerId())) {
+            throw new IllegalAccessException("The user with id: " + oldOwner.getId() + " is not the current owner of the group with id: " + group.getId() + "!");
+        }
+
+        if (newOwner.getId().equals(oldOwner.getId())) {
+            // No ownership change
+            return group;
+        }
+
+        // Change old and new owners additional manager status if appropriate
+        if (!group.getAdditionalManagersUserIds().contains(oldOwner.getId())) {
+            this.groupDatabase.addUserAdditionalManagerList(oldOwner.getId(), group.getId());
+        }
+        if (group.getAdditionalManagersUserIds().contains(newOwner.getId())) {
+            this.groupDatabase.removeUserFromAdditionalManagerList(newOwner.getId(), group.getId());
+        }
+
+        // ! We are mutating this group object, but this particular mutation should be safe !
+        group.setOwnerId(newOwner.getId());
```

Comment:
This is safe because `editUserGroup(group)` transforms the `UserGroupDTO` into a DO to persist it to the DB. `UserGroupDO`s don't store any information about additional managers, so it doesn't matter that we don't update the additional manager list in `group`.

Reply:
Looking at this again - it is a bit of a code smell, as it relies on internal DB logic and what happens inside `editUserGroup`, but provided we don't change how we store additional managers, it will work.

Reply:
I think this is fine. It would be nice if it was more obvious from here that `editUserGroup()` is a simple "map to DO and persist" and therefore other fields like additional managers would be lost anyway, but that's not a problem specific to this change.